### PR TITLE
Llama 2 demo

### PR DESCRIPTION
**Pull request by dkalinowski**

### CODE REVIEW

File: demos/llama_chat/python/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+usage: client.py [-h] --url URL --question QUESTION [--seed SEED] [--actor {general-knowledge,python-programmer}]
+
+Inference script for generating text with llama
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --url URL
+  --question QUESTION
+  --seed SEED
+  --actor {general-knowledge,python-programmer}
```
Comment (mzegla) [inline demos/llama_chat/python/README.md:60]: Can we say more about those two arguments to better guide users?

File: demos/llama_chat/python/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+Examples:
+
+```bash
+python3 client.py --url localhost:9000 --question "How many corners there are in square?" --seed 14140 --actor general-knowledge
+
+ Four. [EOS]%
```
Comment (mzegla) [inline demos/llama_chat/python/README.md:68]: [EOS]% ? Is that some leftover or the client script is supposed to return that?

File: demos/llama_chat/python/client.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            break
+
+
+# python3 client.py --url localhost:11337 --question "How many corners there are in square?" --seed 14140 --actor general-knowledge
+#    Four.
+
+# python3 client.py --url localhost:11337 --question "Write python function to sum 3 numbers." --seed 1332 --actor python-programmer
+#    def sum_three_numbers(a,b,c):
+#      result = a + b + c
+#      return result
```
Comment (mzegla) [inline demos/llama_chat/python/client.py:101]: To be removed?

File: demos/llama_chat/python/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+openvino==2023.1.0.dev20230728
+ovmsclient==2023.0
```
Comment (mzegla) [inline demos/llama_chat/python/requirements.txt:2]:
```suggestion
ovmsclient==2023.1
```

**Review by dtrawins — APPROVED** [review]

File: demos/llama_chat/python/client.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+parser.add_argument('--question', required=True, type=str)
+parser.add_argument('--seed', required=False, type=int, default=0)
+parser.add_argument('--actor', required=False, type=str, choices=['general-knowledge', 'python-programmer'], default='general-knowledge')
+args = parser.parse_args()
+
+np.random.seed(args.seed)
+client = make_grpc_client(args.url)
+
+GENERAL_PRE_PROMPT = """You are super-human who is able to answer any question. All questions and answers should end with [EOS].
+Question: What is the capital of Poland? [EOS]
```
Comment (atobiszei) [inline demos/llama_chat/python/client.py:20]: Why is this EOS needed? Looks like workaround for some issue/limitation.

File: demos/llama_chat/python/client.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+Answer: Warsaw. [EOS]
+
+Question: <Q> [EOS]
+Answer:"""
+
+PROGRAMMER_PRE_PROMPT = """You are super-programmer who is brilliant python coder. All the answers contain code snippets ending with [EOS].
+Question: Write short python function to sum numbers in a list. [EOS]
```
Comment (atobiszei) [inline demos/llama_chat/python/client.py:27]: Do we need to post example question/answer each time?

File: demos/llama_chat/python/README.md (hunk 1 above)
Comment (dkalinowski) [reply demos/llama_chat/python/README.md:60]: done

File: demos/llama_chat/python/README.md (hunk 2 above)
Comment (dkalinowski) [reply demos/llama_chat/python/README.md:68]: This is a method to force model to signal end of sequence

File: demos/llama_chat/python/client.py (hunk 3 above)
Comment (dkalinowski) [reply demos/llama_chat/python/client.py]: removed

File: demos/llama_chat/python/requirements.txt (hunk 4 above)
Comment (dkalinowski) [reply demos/llama_chat/python/requirements.txt]: done

File: demos/llama_chat/python/client.py (hunk 5 above)
Comment (dkalinowski) [reply demos/llama_chat/python/client.py:20]: This is a method to force model to signal end of sequence

File: demos/llama_chat/python/client.py (hunk 6 above)
Comment (dkalinowski) [reply demos/llama_chat/python/client.py:27]: This script does not support chatting, only answers given question one time

**Review by mzegla — APPROVED** [review]

**Pull request merged**